orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -107,7 +107,9 @@ def _forward(self, *inputs, **kwargs):
             Finally, we instantiate the ONNX Runtime InferenceSession.
             '''
             # TODO: using pytorch for evaluation for now. We will use ORT for evaluation later.
-            if not self._is_training:
+            # TODO: If the model is being executed with the gradient disabled (inside torch.no_grad() context for example),
+            # leverage pytorch model for now.
+            if not self._is_training or not torch.is_grad_enabled():
```

Comment:
Instead of having 2 places to indicate the model is in training mode, we can upgrade `self._is_training` to a `@property` and in the getter implementation we add the `torch.is_grad_enabled()`, having a single point of check

Reply:
yeah that is a good idea. I'll do that.